com.unity.mobile.notifications/Runtime/iOS/iOSNotificationsWrapper.cs

```diff
@@ -169,63 +156,53 @@ public static iOSNotificationSettings GetNotificationSettings()
         public static void ScheduleLocalNotification(iOSNotificationData data)
         {
 #if UNITY_IOS && !UNITY_EDITOR
-            IntPtr ptr = Marshal.AllocHGlobal(Marshal.SizeOf(data));
-            Marshal.StructureToPtr(data, ptr, false);
-
-            _ScheduleLocalNotification(ptr);
+            _ScheduleLocalNotification(data);
 #endif
         }
 
         public static iOSNotificationData[] GetDeliveredNotificationData()
         {
 #if UNITY_IOS && !UNITY_EDITOR
-            var size = _GetDeliveredNotificationDataCount();
-
-            var dataList = new List<iOSNotificationData>();
-            for (var i = 0; i < size; i++)
-            {
-                iOSNotificationData data;
-                IntPtr ptr = _GetDeliveredNotificationDataAt(i);
-
-                if (ptr != IntPtr.Zero)
-                {
-                    data = (iOSNotificationData)Marshal.PtrToStructure(ptr, typeof(iOSNotificationData));
-                    dataList.Add(data);
-                    _FreeUnmanagediOSNotificationData(ptr);
-                }
-            }
-
-            return dataList.ToArray();
+            int count;
+            var ptr = _GetDeliveredNotificationDataArray(out count);
+            return MarshalAndFreeNotificationDataArray(ptr, count);
 #else
-            return new iOSNotificationData[] {};
+            return null;
 #endif
         }
 
         public static iOSNotificationData[] GetScheduledNotificationData()
         {
 #if UNITY_IOS && !UNITY_EDITOR
-            var size = _GetScheduledNotificationDataCount();
+            int count;
+            var ptr = _GetScheduledNotificationDataArray(out count);
+            return MarshalAndFreeNotificationDataArray(ptr, count);
+#else
+            return null;
```

Comment:
Here we return a null instead of an empty array if scheduled notification data is empty. Not sure how this would impact the user code, as previously users can only check if the length is 0, but now they have to check if it's null before checking the length...

Same to GetDeliveredNotificationData.

Reply:
These are internal methods. I've handled null cases where they are called.